packages/preconfirmation-p2p/crates/service/src/lib.rs

```diff
@@ -0,0 +1,739 @@
+        _from: PeerId,
+        _msg: GetRawTxListResponse,
+        _request_id: Option<u64>,
+    ) {
+    }
+    /// Called when a response to a head request is received.
+    fn on_head_response(
+        &self,
+        _from: PeerId,
+        _head: preconfirmation_types::PreconfHead,
+        _request_id: Option<u64>,
+    ) {
+    }
+    /// Called when an inbound request for commitments is received.
+    fn on_inbound_commitments_request(&self, _from: PeerId) {}
+    /// Called when an inbound request for a raw transaction list is received.
+    fn on_inbound_raw_txlist_request(&self, _from: PeerId) {}
+    /// Called when an inbound request for the preconfirmation head is received.
+    fn on_inbound_head_request(&self, _from: PeerId) {}
+    /// Called when a new peer connects.
+    fn on_peer_connected(&self, _peer: PeerId) {}
+    /// Called when a peer disconnects.
+    fn on_peer_disconnected(&self, _peer: PeerId) {}
+    /// Called when a network-level error occurs.
+    fn on_error(&self, _err: &NetworkError) {}
+}
+
+/// Owned service wrapper.
+///
+/// It spawns the libp2p driver onto the tokio runtime and provides
+/// convenience helpers to publish gossip, send requests, and consume events.
+pub struct P2pService {
+    /// Sender for `NetworkCommand`s to the `NetworkDriver`.
+    command_tx: mpsc::Sender<NetworkCommand>,
+    /// Broadcast sender for fanned-out `NetworkEvent`s.
+    events_tx: broadcast::Sender<NetworkEvent>,
+    /// Default subscription used by `next_event()`.
+    primary_rx: broadcast::Receiver<NetworkEvent>,
+    /// Sender for the shutdown signal to the `NetworkDriver` task.
+    shutdown_tx: Option<oneshot::Sender<()>>,
+    /// Handle to the `NetworkDriver`'s background task.
+    join_handle: Option<JoinHandle<()>>,
+    /// Handle to the fanout task copying driver events into the broadcast channel.
+    fanout_handle: Option<JoinHandle<()>>,
+}
+
+impl P2pService {
+    /// Starts the P2P service by constructing a `NetworkDriver` and spawning it
+    /// as a background task on the tokio runtime.
+    ///
+    /// This function initializes the entire network stack and provides a handle
+    /// for sending commands and receiving events.
+    ///
+    /// # Arguments
+    ///
+    /// * `config` - The `NetworkConfig` used to configure the underlying network driver.
+    ///
+    /// # Returns
+    ///
+    /// A `Result` which is `Ok(Self)` on successful startup, or an `anyhow::Error`
+    /// if the network driver fails to initialize.
+    pub fn start(config: NetworkConfig) -> Result<Self> {
+        Self::start_with_lookahead_and_storage(config, None, None)
+    }
+
+    /// Starts the P2P service with an optional lookahead resolver for commitment validation.
+    pub fn start_with_lookahead(
+        config: NetworkConfig,
+        lookahead: Option<std::sync::Arc<dyn LookaheadResolver>>,
+    ) -> Result<Self> {
+        Self::start_with_lookahead_and_storage(config, lookahead, None)
+    }
```

Comment:
Shall we remove these two methods and always enforce users to start with a lookahead provider and a (in-memory) storage?